lighthouse-core/computed/computed-artifact.js

```diff
@@ -30,6 +32,15 @@ function makeComputedArtifact(computableArtifact) {
     const computedName = computableArtifact.name;
 
     const cache = computedCache.get(computedName) || new ArbitraryEqualityMap();
+    if (keys) {
```

Comment:
rather than custom caching based on keys, can this instead rewrite `artifacts` to be an object of just those keys?